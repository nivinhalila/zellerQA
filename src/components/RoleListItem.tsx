import React from "react";
import styled from "styled-components";

import Text from "./Text";

interface LabelProps {
  isSelected: boolean;
}

const Label = styled.label<LabelProps>`
  align-items: center;
  background-color: ${(props) => (props.isSelected ? `#e8f2fb` : `#ffffff`)};
  border-radius: 1rem;
  display: flex;
  flex-grow: 1;
  margin-bottom: 1rem;
  padding: 0rem 2rem;

  &:last-child {
    margin-bottom: 0;
  }
`;

const RadioInput = styled.input`
  height: 2rem;
  margin: 0 1.5rem 0 0;
  width: 2rem;
`;

interface RoleListItemProps {
  label: string;
  isSelected: boolean;
  handleOnChange: () => void;
}

export function RoleListItem(props: RoleListItemProps) {
  const { label, isSelected, handleOnChange } = props;

  return (
    <Label isSelected={isSelected} htmlFor={label}>
      <RadioInput
        data-testid={label}
        checked={isSelected}
        id={label}
        onChange={handleOnChange}
        name="role"
        type="radio"
      />
      <Text>{label}</Text>
    </Label>
  );
}

export default RoleListItem;
