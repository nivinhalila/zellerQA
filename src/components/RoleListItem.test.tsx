import "jest-styled-components";
import { render, fireEvent } from "@testing-library/react";

import RoleListItem from "./RoleListItem";

test("renders role list item component", () => {
  expect.assertions(3);

  const label = "Testing";
  const isSelected = true;
  const handleOnChange = jest.fn();

  const { container, getByTestId, getByText } = render(
    <RoleListItem
      label={label}
      isSelected={isSelected}
      handleOnChange={handleOnChange}
    />
  );

  const input = getByTestId("Testing") as HTMLInputElement;

  expect(getByText("Testing")).toBeInTheDocument();
  expect(input.checked).toEqual(true);
  expect(container.firstChild).toMatchSnapshot();
});

test("input is unselected when is selected is false", async () => {
  expect.assertions(1);

  const label = "Testing";
  const isSelected = false;
  const handleOnChange = jest.fn();

  const { getByTestId } = render(
    <RoleListItem
      data-testid={label}
      label={label}
      isSelected={isSelected}
      handleOnChange={handleOnChange}
    />
  );

  const input = getByTestId("Testing") as HTMLInputElement;

  expect(input.checked).toEqual(false);
});

test("calls handle on change when input changed", () => {
  expect.assertions(1);

  const label = "Testing";
  const isSelected = false;
  const handleOnChange = jest.fn();

  const { getByText } = render(
    <RoleListItem
      label={label}
      isSelected={isSelected}
      handleOnChange={handleOnChange}
    />
  );

  fireEvent.click(getByText(/Testing/));

  expect(handleOnChange).toBeCalledTimes(1);
});
