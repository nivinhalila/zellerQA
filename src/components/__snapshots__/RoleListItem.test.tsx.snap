// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders role list item component 1`] = `
.c2 {
  color: #212322;
  font-size: 1.5em;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #e8f2fb;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 1rem;
  padding: 0rem 2rem;
}

.c0:last-child {
  margin-bottom: 0;
}

.c1 {
  height: 2rem;
  margin: 0 1.5rem 0 0;
  width: 2rem;
}

<label
  class="c0"
  for="Testing"
>
  <input
    checked=""
    class="c1"
    data-testid="Testing"
    id="Testing"
    name="role"
    type="radio"
  />
  <p
    class="c2"
  >
    Testing
  </p>
</label>
`;
